### PR TITLE
Build yocto: cleanup

### PR DESCRIPTION
**Pull request by quaresmajose**

### CODE REVIEW

File: .github/workflows/build-yocto.yml [hunk 1]
```diff
@@ -7,18 +7,50 @@ on:
...
+        with:
+          fetch-depth: 0
+
+      - name: Run kas lock
+        run: |
+          kas dump --update --lock --inplace ci/base.yml
+
+      - name: Prepare artifact
+        run: |
+          mkdir ci/lock
```
Comment (ndechesne) [inline .github/workflows/build-yocto.yml:23]: why is that needed? Don't you just need ci/base.lock.yml?

File: .github/workflows/build-yocto.yml [hunk 2]
```diff
@@ -76,8 +76,9 @@ jobs:
 
       - name: Kas build
         run: |
-          export DL_DIR=/srv/gh-runners/quic-yocto/downloads
-          export SSTATE_DIR=/srv/gh-runners/quic-yocto/sstate-cache
+          export PERSISTENT_DIR=/srv/gh-runners/quic-yocto
```
Comment (ndechesne) [inline .github/workflows/build-yocto.yml:79]: I don't think you need PERSISTENT_DIR anymore. it was used before we agreed to use github artifacts for the lock files, no?

File: .github/workflows/build-yocto.yml (hunk 1 above)
Comment (quaresmajose) [reply .github/workflows/build-yocto.yml:23]: this needs to be rebased but I will wait for the merge of the https://github.com/quic-yocto/meta-qcom-hwe/pull/36

File: .github/workflows/build-yocto.yml (hunk 2 above)
Comment (quaresmajose) [reply .github/workflows/build-yocto.yml:66]: it is not strictly necessary but simplify the variables definition and is there to be used if we need to store something not using the artifacts.

File: .github/workflows/build-yocto.yml [hunk 3]
```diff
@@ -63,10 +63,9 @@ jobs:
...
+          export DL_DIR=${PERSISTENT_DIR}/downloads
+          export SSTATE_DIR=${PERSISTENT_DIR}/sstate-cache
           mkdir -p $DL_DIR
           mkdir -p $SSTATE_DIR
-          mkdir build
-          cd build
-          kas build ../ci/${{ matrix.machine }}.yml
+          kas build ci/${{ matrix.machine }}.yml
```
Comment (ricardosalveti) [inline .github/workflows/build-yocto.yml:71]: I'm not a fan of of building as part of the main git repo, maybe remove but set KAS_BUILD_DIR to a folder outside the repo?

Comment (quaresmajose) [reply .github/workflows/build-yocto.yml:72]: done

File: .github/workflows/build-yocto.yml [hunk 4]
```diff
@@ -63,10 +63,10 @@ jobs:
 
       - name: Kas build
         run: |
-          export DL_DIR=/srv/gh-runners/quic-yocto/downloads
-          export SSTATE_DIR=/srv/gh-runners/quic-yocto/sstate-cache
+          export PERSISTENT_DIR=/srv/gh-runners/quic-yocto
+          export DL_DIR=${PERSISTENT_DIR}/downloads
+          export SSTATE_DIR=${PERSISTENT_DIR}/sstate-cache
+          export KAS_BUILD_DIR=$(realpath $PWD/..)/build
```
Comment (ricardosalveti) [inline .github/workflows/build-yocto.yml:69]: I think it is actually better to set KAS_WORK_DIR instead (KAS_BUILD_DIR is ${KAS_WORK_DIR}/build), that way it also clones the other repos outside the folder where kas build gets called (which is this repo).

Comment (quaresmajose) [reply .github/workflows/build-yocto.yml]: ok, I moved it to the outside of the repo `../kas`, one dir node above.

**Comment by quaresmajose** [comment]
Also changed from PERSISTENT_DIR to CACHE_DIR because the first is already part of bitbake.

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**